### DISCUSSION TITLE
How are free products and returned products typically accounted for in 1) RFM and in 2) CLV?

### DISCUSSION THREAD

**Discussion by hmanz** [pymc-labs/pymc-marketing#682] [Q&A]
In RFM segmentation, I was thinking it makes sense to count free products as affecting recency and frequency, but not monetary. And for returns I think it makes sense to just subtract from frequency and monetary, while updating recency reflect the date of the return. How do people typically treat free products and returned products in RFM segmentation analysis?

In CLV, since recency, frequency, and monetary all have different definitions from what is used in RFM segmentation, I was also wondering what the typical treatment of free products and returned products is. Thank you!

As a side note, I noticed in [the code](https://github.com/pymc-labs/pymc-marketing/blob/61da9c05721a70cd80e41dd68a7772622ca23cac/pymc_marketing/clv/utils.py#L490) that a differentiation is made for the definitions depending on if we are doing RFM segmentation or CLV. However in the docs (e.g. [CLV Quickstart](https://www.pymc-marketing.io/en/stable/notebooks/clv/clv_quickstart.html)) this differentiation is not mentioned and it can lead to some confusion particularly for people who are new to either RFM segmentation or CLV. Should I create an issue for adding it to the docs or is this the right place to mention it? Thanks

**Comment by ColtAllen** [comment]
Hey @hmanz,

> How do people typically treat free products and returned products in RFM segmentation analysis?

I'm curious to hear thoughts from others, but generally promotional giveaways and returns are excluded because they aren't true purchasing activities. It will also skew monetary value calculations based on average transaction values.

> Should I create an issue for adding it to the docs or is this the right place to mention it?

Please do so (and thanks in advance!) [There is a PR open right now for an RFM Segmentation utility](https://github.com/pymc-labs/pymc-marketing/pull/680), but for brevity it does not include any notebook revisions. Also feel free to add any comments/questions to the open PR.

**Reply by hmanz** [reply]
Hey [@ColtAllen](https://github.com/ColtAllen), thank you for your response!

> I'm curious to hear thoughts from others, but generally promotional giveaways and returns are excluded because they aren't true purchasing activities.

Would it depend on the use case? For example it could be more important to know if someone had any recent interaction with the company, rather than it necessarily being a purchase. Or it could be important to subtract returns from user's monetary value because they didn't actually end up giving any money to the company.

> It will also skew monetary value calculations based on average transaction values.

Wouldn't this only be true if you are using average monetary instead of total monetary for M?

> Please do so (and thanks in advance!)

No worries, will do, and thank you!

**Comment by ColtAllen** [comment]
@hmanz,

My apologies - instead of replying to your last message two weeks ago, I edited it into my response haha. I've restored your previous message, and the RFM Segmentation PR is now merged and will be available in the next release.

>> generally promotional giveaways and returns are excluded because they aren't true purchasing activities.

> Would it depend on the use case?

Yes; best to engage with the business on this for specifics.

> >It will also skew monetary value calculations based on average transaction values.

> Wouldn't this only be true if you are using average monetary instead of total monetary for M?

Yes, but using total monetary values will correlate M with F and make it more difficult to differentiate between big spenders and frequent buyers. This will also make the `GammaGammaModel` completely unusable.
